torch_geometric/nn/conv/heat_pyg.py

```diff
@@ -0,0 +1,187 @@
+import torch
+import torch.nn as nn
+
+def retrieve_masks(node_or_edge_type, wanted_types=[]):
+    assert set(wanted_types) < set(node_or_edge_type)
+    return [True if x in wanted_types else False for x in node_or_edge_type]
+
+def flatten_a_list_of_list(list_of_list):
+    flat_list = []
+    for l in list_of_list:
+        flat_list += l
+    return flat_list
+
+class HEATlayer(nn.Module):
+    ''' 
+        1.  type-specific transformation for nodes of different types: ('n1', 'n2').
+            transform nodes from different vector space to the same vector space.
+        2.  edges are assumed to have different types but contains the same attributes.            
+    '''
+    def __init__(self,  node_type_set = ('n1', 'n2', 'n3'), edge_type_set = ('e1', 'e2', 'e3', 'e4'),
```

Comment:
I don't think we need the exact names of node/edge types here. `num_node_types` and `num_edge_types` seems to be sufficient.

Reply:
Yes! Thank you for this comment, I revised the code accordingly, and it works! thank you!